chef/cookbooks/crowbar/templates/default/crowbar-reload-nova-after-upgrade.sh.erb

```diff
@@ -27,21 +27,41 @@ if [[ -f $UPGRADEDIR/crowbar-reload-nova-after-upgrade-ok ]] ; then
     exit 0
 fi
 
+# Check the python-nova package to make sure it is at the latest version with
+# no more recent versions in any zypper repositories (this package can be used
+# as a proxy for making sure all Nova service packages are up to date since
+# they are version locked to python-nova):
+
+if zypper lu -a | grep -w python-nova; then
+  echo "There are available update candidates for python-nova. This means that "
+  echo "the nova service packages on this node are not at the latest possible "
+  echo "version. Please update them and restart the upgrade."
+
+  exit 1
+fi
+
 <% if @nova_controller %>
 for service in conductor scheduler novncproxy serialproxy api; do
     fullname="openstack-nova-$service"
-    if systemctl --quiet is-active $fullname 2>/dev/null ; then
-        systemctl restart $fullname
-    fi
+    systemctl start $fullname
+done
+
+# Apache was also stopped by crowbar-stop-nova-services.sh
+systemctl start apache2
+
+  <% if @remotes_present and @is_cluster_founder %>
+# Start nova-compute on remote nodes:
+for remote in $(crm resource list | awk '$1 ~ /^remote-/  {print $1}'); do
```

Comment:
Why not use `remote_nodes` varaible?

Reply:
Because I'd be building the resource name from the host name in there and a prepended `remote-`. I'm not comfortable assuming that this is always the correct resource name. For instance, what happens if there's a FQDN in the list and a short host name in pacemaker (or vice versa)? I can rely on whatever matches that regex in the pacemaker resource list being a remote resource since pacemaker is the canonical source of resource names.

Reply:
OK